main.ts

```diff
@@ -13,6 +13,24 @@ import * as fs from 'fs'
 const flavor = core.getInput('flavor')
 const architecture = core.getInput('architecture')
 
+/**
+ * Some Azure VM types have a temporary disk which is local to the VM and therefore provides
+ * _much_ faster disk IO than the OS Disk (or any other attached disk).
+ *
+ * Hosted GitHub Actions runners also leverage this disk and do their work in D:/a/_work, so let's
+ * use it too if we can. It leads to a ~25% speed increase when doing heavy IO operations.
+ *
+ * https://learn.microsoft.com/en-us/azure/virtual-machines/managed-disks-overview#temporary-disk
+ */
+function getDriveLetterPrefix(): string {
+  if (fs.existsSync('D:/')) {
+    core.info('Found a fast, temporary disk on this VM (D:/). Will use that.')
+    return 'D:/'
+  }
+
+  return 'C:/'
+}
+
```

Comment:
We could also leverage the `$RUNNER_TEMP` environment variable which resolves to `D:\a\_temp` for example. It depends on the directory where the Actions Runner is installed (e.g. `D:\a` or `C:\a`), meaning that we wouldn't have to use our own logic for this. WDYT? Any concerns about long path issues or something? I think `D:\a\_temp\git-sdk-64-full` vs `D:\git-sdk-64-full` isn't too much of a difference.

[From the docs](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables):

> The path to a temporary directory on the runner. This directory is emptied at the beginning and end of each job. Note that files will not be removed if the runner's user account does not have permission to delete them. For example, `D:\a\_temp`

That should also allow us to get rid of our custom cleanup logic 👀🔥

Reply:
My intuition is that using `$RUNNER_TEMP` could potentially cause problems, even inadvertent ones, because the `TEMP` part of the name suggests something ethereal, not-so-critical.

Long paths are not my concern here, either. But disk space is. I'd love to have the speed improvements, but we need to be sure that we're not accidentally running out of space. According to [this documentation](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) we have 14GB of SSD, which should be enough, even if the SDK weighs in with over 2GB.

Reply:
I just checked and [it seems that there is still a bit over 12GB free](https://github.com/dscho/setup-git-for-windows-sdk/actions/runs/3824869835/jobs/6507389346#step:2:9), so we're good to go!